pizza_cutter_metadetect/masks.py

```diff
@@ -0,0 +1,450 @@
+    msk_img[
+        slice_bnds["min_row"]+srow:slice_bnds["max_row"]+srow,
+        slice_bnds["min_col"]+scol:slice_bnds["max_col"]+scol,
+    ] |= MASK_NOSLICE
+
+
+def make_mask(
+    *,
+    preconfig,
+    missing_slice_inds,
+    obj_data,
+    central_size,
+    buffer_size,
+    wcs,
+    position_offset,
+    coadd_dims,
+    info,
+    gaia_stars=None,
+    healpix_nside=131072,
+):
+    """Make the healsparse mask indicating where measurements were made.
+
+    Parameters
+    ----------
+    preconfig : dict
+        Proprocessing configuration. Must contain gaia_star_masks
+        entry if gaia_stars is not None.
+    missing_slice_inds : list of int
+        The indices into `obj_data` indicating slices that did not get measurements.
+    obj_data : structure numpy array
+        The object data from the MEDS file. Used for getting the slice locations
+        orig_start_col/orig_start_row. These values are the same across bands and
+        so the data from any of the bands is fine.
+    central_size : int
+        The size of the central region of the slice.
+    buffer_size : int
+        The size of the buffer around each slice.
+    wcs : esutil.wcsutil.WCS
+        A WCS object for the coadd tile.
+    position_offset : int
+        The position offset to convert from 0-indexed pixel coordinates into
+        the convention expected by the `wcs`.
+    coadd_dims : 2-tuple of ints
+        The dimension of the coadd image.
+    info : dict
+        The coadd tile info produced by the pizze cutter and stored in the MEDS
+        metadata as the 'tile_info' entry. This dict is used to get the coadd
+        tile unique ra/dec bounds (uramin, uramax, udecmin, udecmax) and if the
+        tile cross ra = 0 (crossra0).
+    gaia_stars : structured numpy array, optional
+        The GAIA star data indicating where the GAIA star mask holes are. The default
+        of None will result in no GAIA star mask holes.
+    healpix_nside : int, optional
+        The nisde of the healsparse mask. The default is 131072.
+
+    Returns
+    -------
+    hs_msk : healsparse map
+        The healsparse boolean map indicating what areas were masked and why. The
+        possible flag values are
+
+            MASK_NOSLICE = 2**0 - indicates no data for a given coadd slice
+            MASK_GAIA_STAR = 2**1 - indicates regions where a GAIA star hole exists
+    """
+    # We will build a coadd image of the mask bits and then convert them to
+    # healsparse at the end. We do things this way for a few reasons.
+    # 1. The coadd image of mask bits is the mask actually used by the code.
+    # 2. We do not have pure geometric primitives that are easy to write down
+    #    in healsparse due to various factors.
+    # 3. it makes the code a bit simpler since we don't have to combine more
+    #    than one representation of the mask to get the final one
+
+    # this image holds the mask bits
+    msk_img = np.zeros(coadd_dims, dtype=np.int32)
```

Comment:
This might be handled below by the `build_base_tile_mask`, but wouldn't it make sense to set all these pixels to `MASK_INTILE = 2**0` because everything in the coadd dims defines the tile boundaries?  Then I think that `build_base_tile_mask` would just need to make an empty healsparse map, and the `update_values_pos` would just work.  (I could be wrong for some reason!)

Reply:
This would work, but computing the ra,dec values for every pixel in a coadd is really slow.

Reply:
Ah, I see.  This way you're only doing the computation for the masked pixels, and using healsparse to fill in the rest.  Makes sense!

Reply:
(In case it's relevant in the future, in the Rubin stack with those WCSs it's possible to efficiently create a full grid of ra/dec within a bounding box).

Reply:
Right. I have WCS interpolation tools around as well. I could/should switch to that to make this simpler maybe.

Reply:
I changed to using the grid directly.